File: Forth_Interpreter/ForthCommands.cpp
#include "ForthCommands.hpp"

GeneralData::GeneralData (std::stack <int> *stack, std::string commandText, 
                 std::string *commandsForSecondProcessing, std::istream *in, std::ostream *out) 
                 : Stack(stack), CommandText(commandText), 
                   CommandsForSecondProcessing(commandsForSecondProcessing),
                   Input(in), Output(out) {}

void ForthCommands::AddGeneralData (GeneralData gendata) {
    GeneralDataForExecutors = gendata;
}

File: Forth_Interpreter_On_Java/src/main/java/ru/nsu/yurakov/Forth/ForthCommands/ForthCommands.java
package ru.nsu.yurakov.Forth.ForthCommands;

import ru.nsu.yurakov.Forth.ForthRuntimeException.ForthRuntimeException;
import ru.nsu.yurakov.Forth.GeneralData.GeneralData;

public abstract class ForthCommands {
    public ForthCommands() {}

    /** Executing the invoked command
     * @return true if something was output to out else false
     * @throws ForthRuntimeException If something went wrong during the execution of the command
     */
    public abstract boolean Run() throws ForthRuntimeException;

    /** Uploading basic data to the executor
     * @param gendata Data to be copied
     */
    public void AddGeneralData (GeneralData gendata) {
        GeneralDataForExecutors = gendata;
    }

    protected GeneralData GeneralDataForExecutors;
}